### ISSUE TITLE
bug report - bdc --download breaks when adding azure only notebooks

### ISSUE THREAD

**Issue by juno-lee** [databricks-edu/build-tooling#91]
When I remove the lines `only_in_profile: azure` from the notebooks section (lines 196, 200, 204 in the attached build.yaml file), bdc --download works properly. However, when I add those lines back in, I get the following warnings:

```txt
WARNING: Cannot find downloaded version of course notebook "/Users/juno.lee/repos/training/modules/DB-105/Azure-SQL-Data-Warehouse.scala".
WARNING: Cannot find downloaded version of course notebook "/Users/juno.lee/repos/training/modules/DB-105/Azure-Blob-Storage.scala".
WARNING: Cannot find downloaded version of course notebook "/Users/juno.lee/repos/training/modules/DB-105/Azure-ADLS-Gen-2.scala".
```

```txt
WARNING: These files from /Curriculum/Spark-ILT-105 aren't in the build file and were not copied.
    Reading & Writing Data/Azure 3 - SQL Data Warehouse.scala
    Reading & Writing Data/Azure 1 - Blob Storage.scala
    Reading & Writing Data/Azure 2 - ADLS Gen 2.scala
```
[build.yaml.zip](https://github.com/databricks-edu/build-tooling/files/3332164/build.yaml.zip)

**Comment by bmc** [comment]
Juno, I cannot reproduce this problem. I used your `build.yaml` to upload notebooks to _trainers_. I left all the `only_in_profile:` lines in place. I then ran a `--download`. It worked fine.

Then, just to be sure, I commented out the `only_in_profile:` lines before doing the upload. I then uncommented them again, and ran a download. Again, it worked fine.

We may have to tag-team debug your setup, because I can't reproduce it here.

**Comment by bmc** [comment]
I can reproduce it. The problem is related to a fix introduced in commit dfa2e02:

> Fixed bug relating to upload and download capability: If two notebooks with separate profiles (“amazon” and “azure”) map to the same dest value, bdc would overwrite one of them during the upload and would arbitrarily choose one on the download.

Basically, the fix addressed this situation:

```yaml
- src: AzureLesson01.scala
  dest: "${target_lang}/Lesson-01.$target_ext"
- src: AmazonLesson01.scala
  dest: "${target_lang}/Lesson-01.$target_ext"
```

So, with the above, on upload:

- `bdc` would first upload `AzureLesson01.scala` to `Lesson-01.scala` (because it tries to mirror the _target_ file layout and naming)
- It would _then_ upload `AmazonLesson01.scala` to `Lesson-01.scala`, overwriting the Azure version.

On download, it would only find a single `Lesson-01.scala`, and it would (a) overwrite (probably) the Azure one on the local file system, and then complain it couldn't find the Amazon one. Or it would overwrite both with the Azure version. (I don't remember.)

The solution: On upload, the target file name (`Lesson-01.scala` in this case) is _prepended_ with a profile abbreviation, "az" or "am". Thus, with that fix in place on upload:

- `AzureLesson01.scala` gets uploaded as `az-Lesson-01.scala`
- `AmazonLesson01.scala` gets uploaded as `am-Lesson-01.scala`

On download, `bdc` expects the same naming convention to apply.

In this case, your setup is breaking that naming convention. `bdc` expects to find a notebook called `Reading & Writing Data/az-Azure 1 - Blob Storage.scala` on the shard, which it would then download and save as `.../modules/DB-105/Azure-SQL-Data-Warehouse.scala`. But you didn't call the file `az-Azure 1 - Blob Storage.scala`. You called it `Azure 1 - Blob Storage.scala`.

I'm not sure how to fix this without breaking other things.

The workaround is to adhere to the naming conventions, namely, **up in the shard**, put "az-" in front of the notebook name, exactly as `--upload` would have done if you'd uploaded it.

**Comment by bmc** [comment]
Per a discussion with @juno-lee, I won't be fixing this one.
